### PR TITLE
Add Leave Group functionality and ran auto code reformater

### PR DESCRIPTION
**Pull request by AAAstorga**
A lot of files were changed bc IntelliJ linted them and did some housekeeping. It was actually pretty nice.

### CODE REVIEW

File: app/src/main/java/com/cornellappdev/android/pollo/NetworkUtils.java [hunk 1]
```diff
@@ -12,30 +11,22 @@
 import com.cornellappdev.android.pollo.Models.User;
 import com.cornellappdev.android.pollo.Models.UserSession;
 import com.google.gson.Gson;
-
+import okhttp3.*;
```
Comment (ningning621) [inline app/src/main/java/com/cornellappdev/android/pollo/NetworkUtils.java]: you're sure that you want to import everything from okhttp3?

File: app/src/main/java/com/cornellappdev/android/pollo/PollGroupActivity.java [hunk 2]
```diff
@@ -171,9 +107,50 @@ public void run() {
                     JSONObject data = (JSONObject) args[0];
                     try {
                         userCount = data.getInt("count");
-                    } catch (JSONException e) {}
+                    } catch (JSONException e) {
```
Comment (ningning621) [inline app/src/main/java/com/cornellappdev/android/pollo/PollGroupActivity.java]: maybe drop in an `e.printStackTrace()` or the kotlin equivalent?

File: app/src/main/java/com/cornellappdev/android/pollo/NetworkUtils.java (hunk 1 above)
Comment (young-k) [reply app/src/main/java/com/cornellappdev/android/pollo/NetworkUtils.java]: ^ we should avoid this, i agree

File: app/src/main/java/com/cornellappdev/android/pollo/NetworkUtils.java [hunk 3]
```diff
@@ -12,30 +11,22 @@
...
-import okhttp3.ResponseBody;
-
 final class NetworkUtils {
 
     static final String AUTHORIZATION = "Authorization";
     static final String BEARER = "Bearer";
-    private static final OkHttpClient client = new OkHttpClient();
     static final MediaType JSON = MediaType.get("application/json; charset=utf-8");
-
+    static final String JOIN_GROUP_ROUTE = "/join/session";
```
Comment (young-k) [inline app/src/main/java/com/cornellappdev/android/pollo/NetworkUtils.java]: `private static final`? we should have a modifier

File: app/src/main/java/com/cornellappdev/android/pollo/PollGroupActivity.java [hunk 4]
```diff
@@ -171,9 +107,50 @@ public void run() {
...
+        mSocket.on(USER_POLL_END, onUserPollEnd);
+        mSocket.on(USER_POLL_RESULTS, onUserPollResults);
+        mSocket.on(USER_POLL_RESULTS_LIVE, onUserPollResultsLive);
+        mSocket.on(USER_COUNT, onUserCount);
+        mSocket.connect();
+    }
+
+    @Override
+    public void onItemClick(View view, int position) {
+
```
Comment (young-k) [inline app/src/main/java/com/cornellappdev/android/pollo/PollGroupActivity.java]: maybe add a comment here for a `// TODO`?

File: app/src/main/java/com/cornellappdev/android/pollo/PollGroupActivity.java [hunk 5]
```diff
@@ -171,9 +107,50 @@ public void run() {
...
             });
         }
     };
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        try {
+            mSocket = IO.socket(BuildConfig.BACKEND_URI);
+        } catch (URISyntaxException e) {
```
Comment (young-k) [inline app/src/main/java/com/cornellappdev/android/pollo/PollGroupActivity.java]: just making sure you're fine with doing nothing here?

File: app/src/main/java/com/cornellappdev/android/pollo/PollGroupActivity.java [hunk 6]
```diff
@@ -171,9 +107,50 @@ public void run() {
...
+        }
+
+        @Override
+        public Fragment getItem(int position) {
+            return PollFragment.newInstance(position + 1);
+        }
+
+        @Override
+        public int getCount() {
+            // Show 2 total pages.
```
Comment (young-k) [inline app/src/main/java/com/cornellappdev/android/pollo/PollGroupActivity.java]: prefer having this comment on the previous line

File: app/src/main/java/com/cornellappdev/android/pollo/PollRecyclerView.java [hunk 7]
```diff
@@ -43,6 +41,23 @@ public int getItemCount() {
         return mData.size();
     }
 
+    public void addAll(List<Group> newList) {
+        mData.clear();
+        mData.addAll(newList);
+    }
+
+    Group getItem(int id) {
```
Comment (young-k) [inline app/src/main/java/com/cornellappdev/android/pollo/PollRecyclerView.java]: no visibility modifier on this? i.e. public, etc

File: app/src/main/java/com/cornellappdev/android/pollo/PollRecyclerView.java [hunk 8]
```diff
@@ -6,19 +6,17 @@
 import android.view.View;
 import android.view.ViewGroup;
 import android.widget.TextView;
-
 import com.cornellappdev.android.pollo.Models.Group;
 
 import java.util.List;
 
 public class PollRecyclerView extends RecyclerView.Adapter<PollRecyclerView.ViewHolder> {
+    final static String[] TIME_LABELS = {"years", "months", "weeks", "days",
```
Comment (young-k) [inline app/src/main/java/com/cornellappdev/android/pollo/PollRecyclerView.java]: `private final static`?

File: app/src/main/java/com/cornellappdev/android/pollo/PollRecyclerView.java [hunk 9]
```diff
@@ -43,6 +41,23 @@ public int getItemCount() {
...
+    public void addAll(List<Group> newList) {
+        mData.clear();
+        mData.addAll(newList);
+    }
+
+    Group getItem(int id) {
+        return mData.get(id);
+    }
+
+    void setClickListener(ItemClickListener itemClickListener) {
```
Comment (young-k) [inline app/src/main/java/com/cornellappdev/android/pollo/PollRecyclerView.java]: same q regarding visibility

File: app/src/main/java/com/cornellappdev/android/pollo/PollRecyclerView.java [hunk 10]
```diff
@@ -43,6 +41,23 @@ public int getItemCount() {
...
+        this.mClickListener = itemClickListener;
+    }
+
+    public interface ItemClickListener {
+        void onItemClick(View view, int position);
+    }
+
     public class ViewHolder extends RecyclerView.ViewHolder implements View.OnClickListener {
         final TextView pollOption;
         final TextView pollOptionMenu;
```
Comment (young-k) [inline app/src/main/java/com/cornellappdev/android/pollo/PollRecyclerView.java]: should this be `mPoll..`?

File: app/src/main/java/com/cornellappdev/android/pollo/Util.java [hunk 11]
```diff
@@ -5,6 +5,17 @@
     //Average number of seconds in years, months, weeks, days, hours, and minutes.
     static final int[] TIME_BLOCKS_IN_SECONDS = {31536000, 2628000, 604800, 86400, 3600, 60};
 
+    public static int[] splitToComponentTimes(long time) {
+        int[] dates = new int[7];
+        for (int i = 0; i < TIME_BLOCKS_IN_SECONDS.length; i++) {
+            dates[i] = (int) time / TIME_BLOCKS_IN_SECONDS[i];
+            time = (int) time % TIME_BLOCKS_IN_SECONDS[i];
+        }
+        dates[6] = (int) time;
```
Comment (young-k) [inline app/src/main/java/com/cornellappdev/android/pollo/Util.java]: why not have this function take `int time` and have the casting done by the caller?

File: app/src/main/res/layout/manage_group_view.xml [hunk 12]
```diff
@@ -38,18 +39,47 @@
...
+            android:layout_marginStart="10dp"
+            android:layout_marginTop="-2.5dp"
+            android:layout_width="45dp"
+            android:text="Leave"
+            android:textColor="@color/red"
+            android:textSize="16sp" />
+
+    </LinearLayout>
+
+    <!--<Button-->
```
Comment (young-k) [inline app/src/main/res/layout/manage_group_view.xml]: delete?

File: app/src/main/java/com/cornellappdev/android/pollo/NetworkUtils.java (hunk 1 above)
Comment (AAAstorga) [reply app/src/main/java/com/cornellappdev/android/pollo/NetworkUtils.java]: This was refactored automatically by intellij. Also I am going to ignore fixes for java files bc they will be removed. This file is barely used and will be removed once I finish moving over networking.

**Pull request merged**